File: Analyst/src/Domain/Entities/SocialMediaPlatform.cs
using Domain.Interfaces;
using Domain.Primitives;

namespace Domain.Entities
{
    public class SocialMediaPlatform : Entity<int>,
        IAggregateRoot,
        IAuditableEntity
    {
        public string Name { get; private set; } = string.Empty;
        public string IconUrl { get; private set; } = string.Empty;

        public DateTime CreatedOnUtc { get; set; }
        public DateTime? ModifiedOnUtc { get; set; }

        public ICollection<UserSocialMediaUsage> UsageRecords { get; private set; } = [];
        public ICollection<UsageSummary> UsageSummaries { get; private set; } = [];
        public ICollection<UserUsageGoal> UsageGoals { get; private set; } = [];

        protected SocialMediaPlatform()
        {
        }

        private SocialMediaPlatform(string name,
            string iconUrl)
        {
            Name = name;
            IconUrl = iconUrl;
        }


        public static SocialMediaPlatform Create(string name,
            string iconUrl)
        {
            return new SocialMediaPlatform(name,
                iconUrl);
        }
    }
}

File: Analyst/src/ReactApp/src/entities/SocialMediaPlatform.ts
import { z } from 'zod'
import { UsageSummarySchema } from './UsageSummary'
import { UserSocialMediaUsageSchema } from './UserSocialMediaUsage'
import { UserUsageGoalSchema } from './UserUsageGoal'

export const SocialMediaPlatformSchema = z.object({
    id: z.number().optional(),
    name: z.string().min(1, { message: 'Platform name is required' }),
    iconUrl: z
        .string()
        .url({ message: 'Invalid icon URL' })
        .or(z.literal(''))
        .optional(),

    usageRecords: z.array(UserSocialMediaUsageSchema).optional(),
    usageSummaries: z.array(UsageSummarySchema).optional(),
    usageGoals: z.array(UserUsageGoalSchema).optional(),
})

export type SocialMediaPlatform = z.infer<typeof SocialMediaPlatformSchema>

export function createSocialMediaPlatform(data: unknown): SocialMediaPlatform {
    return SocialMediaPlatformSchema.parse(data)
}

export const SocialMediaPlatformUpdateSchema =
    SocialMediaPlatformSchema.partial()

export type SocialMediaPlatformUpdate = z.infer<
    typeof SocialMediaPlatformUpdateSchema
>

export function updateSocialMediaPlatform(
    platform: SocialMediaPlatform,
    updates: Partial<SocialMediaPlatformUpdate>
): SocialMediaPlatform {
    const merged = { ...platform, ...updates }
    return SocialMediaPlatformSchema.parse(merged)
}
